crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes/any_class_string_like.rs

```diff
@@ -58,180 +35,65 @@ fn get_attribute_name(attribute: &JsxAttribute) -> Option<TokenText> {
     )
 }
 
-fn is_target_attribute(attribute: &JsxAttribute, target_attributes: &[String]) -> bool {
-    match get_attribute_name(attribute) {
-        Some(name) => target_attributes.contains(&name.to_string()),
-        None => false,
-    }
-}
-
-// attributes visitor
-// ------------------
-
-#[derive(Default)]
-struct StringLiteralInAttributeVisitor {
-    in_target_attribute: bool,
+declare_node_union! {
+    /// A string literal, JSX string, or template chunk representing a CSS class string.
+    pub AnyClassStringLike = JsStringLiteralExpression | JsxString | JsTemplateChunkElement
 }
 
-// Finds class-like strings in JSX attributes, including class, className, and others defined in the options.
-// TODO: match object properties too
-impl Visitor for StringLiteralInAttributeVisitor {
-    type Language = JsLanguage;
-    fn visit(
-        &mut self,
-        event: &WalkEvent<SyntaxNode<Self::Language>>,
-        mut ctx: VisitorContext<Self::Language>,
-    ) {
-        let options = get_options_from_analyzer(ctx.options);
-        let attributes = match &options.attributes {
-            Some(attributes) => attributes,
-            None => return,
-        };
-        match event {
-            WalkEvent::Enter(node) => {
-                // When entering an attribute node, track if we are in a target attribute.
-                if let Some(attribute) = JsxAttribute::cast_ref(node) {
-                    self.in_target_attribute = is_target_attribute(&attribute, attributes);
-                }
-
-                // When entering a JSX string node, and we are in a target attribute, emit.
-                if let Some(jsx_string) = JsxString::cast_ref(node) {
-                    if self.in_target_attribute {
-                        ctx.match_query(AnyClassStringLike::JsxString(jsx_string));
+impl AnyClassStringLike {
+    pub(crate) fn should_visit(&self, options: &UtilityClassSortingOptions) -> Option<bool> {
```

Comment:
That's because we need to navigate the CST, and when navigating nodes, transform `Result<>` to `Option<>`. By returning an `Option`, because use the `?` operator to early return possible `None`. This is a practice we use in almost all rules to make the coding better to write, instead of always handling `Result` and `Option` by hand.